### PR TITLE
Features to improve maintainability of plugin in managed service

### PR DESCRIPTION
**Pull request by JohannesDaniel**
### Description
Adds several standard OpenSearch features to improve maintainability of plugin in managed service

### Issues Resolved
https://github.com/opensearch-project/opensearch-learning-to-rank-base/issues/62

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
This is a huge PR. Can you split it into smaller PRs?
Suggest add more details in description of all of the changes in the PR

File: src/test/java/org/opensearch/ltr/utils/StoreUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
+package org.opensearch.ltr.utils;
```
Comment (ylwu-amzn) [inline src/test/java/org/opensearch/ltr/utils/StoreUtilsTest.java:1]: Add license header

**Comment by JohannesDaniel** [comment]
@ylwu-amzn The PR is quite comprehensive, but the commits should be quite well separated:
* Forst commit is for circuit breaker
* Second commit is for settings
* Third commit is for stats

We could split this into three separate PRs, if you consider this to be more suitable, but in general this PR is about adding features in org.opensearch.ltr and applying these features in the actual LTR rest, query, etc. stuff. How about a walkthrough together?

**Comment by kkhatua** [comment]
@noCharger  would you be able to help here with the review ? Within AWS Managed Service, if there are some learnings that can help @JohannesDaniel  make changes accordingly, it would benefit both targeted managed services, with @JohannesDaniel  getting a more robust version of LTR  ready for his managed service, and AWS inheriting a more user-friendly plugin from opensource.

**Comment by anirudha** [comment]
Can we split into smaller prs for review please. 

Thanks. Possible add a design overview the explain the changes that will help review faster

**Comment by JohannesDaniel** [comment]
Closed in favor of smaller PRs.

**Pull request closed without merging**